megamek/src/megamek/server/totalwarfare/TWGameManager.java

```diff
@@ -31253,6 +31268,98 @@ public List<SmokeCloud> getSmokeCloudList() {
+        Vector<Report> vFullReport = new Vector<>();
+        Roll roll = Compute.rollD6(1);
+
+        if (roll.getIntValue() > 0) {
+            Report.addNewline(vFullReport);
+            vFullReport.add(new Report(5620, Report.PUBLIC));
+
+            Roll rollNumber = Compute.rollD6(1);
+            int numberOfStrikes = Math.max(1, rollNumber.getIntValue() / 2);
+
+            for (int i = 0; i < numberOfStrikes; i++) {
+                Roll rollType = Compute.rollD6(1);
+                int damage;
+                switch (rollType.getIntValue()) {
+                    case 1:
+                    case 2:
+                    case 3:
+                        damage = 5;
+                        break;
+                    case 4:
+                    case 5:
+                        damage = 10;
+                        break;
+                    default:
+                        damage = 15;
+                }
+
+                Coords coords;
+
+                if (game.getOptions().booleanOption(OptionsConstants.ADVANCED_LIGHTNING_STORM_TARGETS_UNITS)) {
+                    List<Entity> entities = game.getEntitiesVector();
+                    int index = Compute.randomInt(entities.size());
+                    coords = entities.get(index).getPosition();
+                } else {
+                    int x = Compute.randomInt(game.getBoard().getWidth());
+                    int y = Compute.randomInt(game.getBoard().getHeight());
+                    coords = new Coords(x, y);
+                }
+
+                Report r;
+                r = new Report(5621);
+                r.add(coords.getBoardNum());
+                vFullReport.add(r);
+
+                lightningStormDamage(coords, damage, vFullReport);
+
+                if (rollType.getIntValue() == 6) {
+                    for (Coords locationAdjacent : coords.allAdjacent()) {
+                        if (game.getBoard().getHex(locationAdjacent) != null) {
+                            r = new Report(5622);
+                            r.add(locationAdjacent.getBoardNum());
+                            vFullReport.add(r);
+
+                            lightningStormDamage(locationAdjacent, 5, vFullReport);
+                        }
+                    }
+                }
+            }
+        }
+
+        return vFullReport;
+    }
+
+    private void lightningStormDamage(Coords coords, int damage, Vector<Report> vFullReport) {
+        Vector<Report> newReports = tryClearHex(coords, damage, Entity.NONE);
+        vFullReport.addAll(newReports);
+
+        Building bldg = game.getBoard().getBuildingAt(coords);
+        if (bldg != null) {
+            Vector<Report> buildingReport = damageBuilding(bldg, damage, coords);
+            vFullReport.addAll(buildingReport);
+        }
+
+        List<Entity> hitEntities = game.getEntitiesVector().stream()
+                .filter(e -> e.getPosition().equals(coords)
```

Comment:
getPosition can return null, better turn this comparison around

Reply:
also, should the lightning storm disregard offboard units?

Reply:
why not gun emplacements?

Reply:
by default it only targets a on board hexes.  
adding off board hexes would make hitting anything very rare, since you can have units sheets away from the main board.

if you use the option to target units, it will target off board units also if they have a posiition.

for gun emplacements I was thinking the building takes the damage.  that is what it looks like it does for other damage types.

![image](https://github.com/user-attachments/assets/6e947ad2-6cd6-46c8-9b17-ea913dd44576)
![image](https://github.com/user-attachments/assets/130bc412-d72c-42ae-ab21-bb468081c276)